src/frontend/src/views/Data/components/SamplesView/components/SamplesTable/index.tsx

```diff
@@ -0,0 +1,108 @@
+import { CellBasic, CellHeader, Table, TableHeader, TableRow } from "czifui";
+import {
+  createColumnHelper,
+  flexRender,
+  getCoreRowModel,
+  useReactTable,
+} from "@tanstack/react-table";
+import { IdMap } from "src/common/utils/dataTransforms";
+import { map } from "lodash";
+import { useEffect, useState } from "react";
+
+// TODO-TR (mlila): types
+interface Props {
+  data: IdMap<Sample> | undefined;
+  isLoading: boolean;
+}
+
+const columnHelper = createColumnHelper<Sample>();
+
+const columns = [
+  columnHelper.accessor("privateId", {
+    header: "Private ID",
+  }),
+  columnHelper.accessor("publicId", {
+    header: "Public ID",
+  }),
+  columnHelper.accessor("collectionDate", {
+    header: "Collection Date",
+  }),
+  columnHelper.accessor((obj) => JSON.stringify(obj.lineage), {
+    id: "lineage",
+    header: "Lineage",
+  }),
+  columnHelper.accessor("uploadDate", {
+    header: "Upload Date",
+  }),
+  columnHelper.accessor((obj) => JSON.stringify(obj.collectionLocation), {
+    id: "collectionLocation",
+    header: "Collection Location",
+  }),
+  columnHelper.accessor("sequencingDate", {
+    header: "Sequencing Date",
+  }),
+  columnHelper.accessor((obj) => JSON.stringify(obj.gisaid), {
+    id: "gisaid",
+    header: "GISAID",
+  }),
+];
+
+const SamplesTable = ({ data, isLoading }: Props): JSX.Element => {
+  const [samples, setSamples] = useState<Sample[]>([]);
+
+  useEffect(() => {
+    if (!data) return;
+
+    const newSamples = map(data, (v) => v);
+    setSamples(newSamples);
```

Comment:
I'm not sure I understand the purpose of making a copy of the upstream data here in the SamplesTable. Is it mostly here to convert the `data` from being a lookup dict into being an array so it can become rows of a table?

Reply:
yes